contracts/contracts/strategies/NativeStaking/ValidatorAccountant.sol

```diff
@@ -0,0 +1,269 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import { ValidatorRegistrator } from "./ValidatorRegistrator.sol";
+import { IWETH9 } from "../../interfaces/IWETH9.sol";
+
+/// @title Validator Accountant
+/// @notice Attributes the ETH swept from beacon chain validators to this strategy contract
+/// as either full or partial withdrawals. Partial withdrawals being consensus rewards.
+/// Full withdrawals are from exited validators.
+/// @author Origin Protocol Inc
+abstract contract ValidatorAccountant is ValidatorRegistrator {
+    /// @notice The minimum amount of blocks that need to pass between two calls to manuallyFixAccounting
+    uint256 public constant MIN_FIX_ACCOUNTING_CADENCE = 7200; // 1 day
+    /// @notice The maximum amount of ETH that can be staked by a validator
+    /// @dev this can change in the future with EIP-7251, Increase the MAX_EFFECTIVE_BALANCE
+    uint256 public constant MAX_STAKE = 32 ether;
+
+    /// @notice Keeps track of the total consensus rewards swept from the beacon chain
+    uint256 public consensusRewards;
+
+    /// @notice start of fuse interval
+    uint256 public fuseIntervalStart;
+    /// @notice end of fuse interval
+    uint256 public fuseIntervalEnd;
```

Comment:
These will aways be relatively small numbers and will be read together. Perhaps we could pack them together as `uint128`. This will save a cold storage read during our accounting, which the protocol will be paying for to happen across multiple instances of this strategy.

Reply:
I thought the same but it actually costs more gas. Doing this change the average. For the Holesky fork tests, `doAccounting` average gas costs increased by 642 gas.

The reason is reading the second storage slot with `fuseIntervalEnd` is conditional on `ethRemaining` being greater than `fuseIntervalStart`. Most of the time the `ethRemaining` will be less than `fuseIntervalStart` hence the `ethRemaining > fuseIntervalEnd` condition is not executed.